Framework/Algorithms/src/ConvertSpectrumAxis2.cpp

```diff
@@ -153,13 +149,24 @@ void ConvertSpectrumAxis2::createElasticQMap(API::Progress &progress,
   else if (emodeStr == "Indirect")
     emode = 2;
 
-  auto &spectrumInfo = inputWS->spectrumInfo();
+  const auto &spectrumInfo = inputWS->spectrumInfo();
+  const auto &detectorInfo = inputWS->detectorInfo();
+  const size_t nHist = spectrumInfo.size();
   for (size_t i = 0; i < nHist; i++) {
     double theta(0.0), efixed(0.0);
     if (!spectrumInfo.isMonitor(i)) {
       theta = 0.5 * spectrumInfo.twoTheta(i);
-      efixed =
-          getEfixed(spectrumInfo.detector(i), inputWS, emode); // get efixed
```

Comment:
Oh, right I missed that changed bit in the implementation of `getEfixed`. It is fine then :+1: 

But actually you can then remove one `if`, I think?
https://github.com/mantidproject/mantid/pull/18528/files/e79bb8064b113dc8d02e8e3b8f8b6e26e2e8ad4f#diff-29d57758573e49db6d230ece2344892cR266